agent/src/main/java/io/pyroscope/javaagent/impl/SamplingProfilingScheduler.java

```diff
@@ -0,0 +1,66 @@
+package io.pyroscope.javaagent.impl;
+
+import static io.pyroscope.javaagent.DateUtils.truncate;
+
+import io.pyroscope.javaagent.Profiler;
+import io.pyroscope.javaagent.Snapshot;
+import io.pyroscope.javaagent.api.Exporter;
+import io.pyroscope.javaagent.api.ProfilingScheduler;
+import io.pyroscope.javaagent.config.Config;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class SamplingProfilingScheduler implements ProfilingScheduler {
+
+    private final Config config;
+    private final Exporter exporter;
+
+    private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor(r -> {
+        Thread t = Executors.defaultThreadFactory().newThread(r);
+        t.setName("PyroscopeProfilingScheduler_Sampling");
+        t.setDaemon(true);
+        return t;
+    });
+
+    public SamplingProfilingScheduler(Config config, Exporter exporter) {
+        this.config = config;
+        this.exporter = exporter;
+    }
+
+    @Override
+    public void start(Profiler profiler) {
+        Duration samplingDuration = config.samplingDuration;
+        final Runnable dumpProfile = () -> {
+            Instant profilingStartTime = Instant.now();
+            profiler.start();
+            try {
+                Thread.sleep(samplingDuration.toMillis());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+            profiler.stop();
+
+            Snapshot snapshot = profiler.dumpProfile(profilingStartTime);
```

Comment:
we need profilingStartTime to be truncated to 10s(uploadInterval), see #40

Reply:
Changed to `truncate(profilingStartTime, uploadInterval)`